src/main/java/com/schematica/client/renderer/RenderSchematic.java

```diff
@@ -62,54 +67,56 @@
-    private final Set<RenderOverlay>                    overlaysToUpdate = Sets.newLinkedHashSet();
-    private final ChunkRenderDispatcher   renderDispatcher        = new ChunkRenderDispatcher();
-    private final OverlayRenderDispatcher renderDispatcherOverlay = new OverlayRenderDispatcher();
-    private final BlockPos.MutableBlockPos tmp                      = new BlockPos.MutableBlockPos();
+    private final Set<RenderOverlay>       overlaysToUpdate        = Sets.newLinkedHashSet();
+    private ChunkRenderDispatcher    renderDispatcher        = null;
+    private OverlayRenderDispatcher  renderDispatcherOverlay = null;
+    private final BlockPos.MutableBlockPos tmp                     = new BlockPos.MutableBlockPos();
     @Nullable
     private SchematicWorld world;
     @NotNull
     private Set<RenderChunk>                      chunksToUpdate = Sets.newLinkedHashSet();
     @NotNull
     private List<ContainerLocalRenderInformation> renderInfos    = Lists.newArrayListWithCapacity(CHUNKS);
     @Nullable
-    private ViewFrustumOverlay viewFrustum;
-    private       double                  frustumUpdatePosX       = Double.MIN_VALUE;
-    private       double                  frustumUpdatePosY       = Double.MIN_VALUE;
-    private       double                  frustumUpdatePosZ       = Double.MIN_VALUE;
-    private       int                     frustumUpdatePosChunkX  = Integer.MIN_VALUE;
-    private       int                     frustumUpdatePosChunkY  = Integer.MIN_VALUE;
-    private       int                     frustumUpdatePosChunkZ  = Integer.MIN_VALUE;
-    private       double                  lastViewEntityX         = Double.MIN_VALUE;
-    private       double                  lastViewEntityY         = Double.MIN_VALUE;
-    private       double                  lastViewEntityZ         = Double.MIN_VALUE;
-    private       double                  lastViewEntityPitch     = Double.MIN_VALUE;
-    private       double                  lastViewEntityYaw       = Double.MIN_VALUE;
+    private ViewFrustumOverlay viewFrustum = null;
+    private double frustumUpdatePosX      = Double.MIN_VALUE;
+    private double frustumUpdatePosY      = Double.MIN_VALUE;
+    private double frustumUpdatePosZ      = Double.MIN_VALUE;
+    private int    frustumUpdatePosChunkX = Integer.MIN_VALUE;
+    private int    frustumUpdatePosChunkY = Integer.MIN_VALUE;
+    private int    frustumUpdatePosChunkZ = Integer.MIN_VALUE;
+    private double lastViewEntityX        = Double.MIN_VALUE;
+    private double lastViewEntityY        = Double.MIN_VALUE;
+    private double lastViewEntityZ        = Double.MIN_VALUE;
+    private double lastViewEntityPitch    = Double.MIN_VALUE;
+    private double lastViewEntityYaw      = Double.MIN_VALUE;
     private AbstractSchematicChunkRenderContainer renderContainer;
     private int renderDistanceChunks = -1;
     private int countEntitiesTotal;
+    private int countTileEntitiesTotal;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "countTileEntitiesTotal" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1068)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the "countTileEntitiesTotal" field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1450)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Perhaps 'countTileEntitiesTotal' could be replaced by a local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ASingularField)